lib/Dialect/FIRRTL/FIRRTLDialect.cpp

```diff
@@ -21,19 +21,9 @@ using namespace firrtl;
 // Dialect specification.
 //===----------------------------------------------------------------------===//
 
-// If the specified attribute set contains the firrtl.name attribute, return it.
-StringAttr firrtl::getFIRRTLNameAttr(ArrayRef<NamedAttribute> attrs) {
-  for (auto &argAttr : attrs) {
-    // FIXME: We currently use firrtl.name instead of name because this makes
-    // the FunctionLike handling in MLIR core happier.  It otherwise doesn't
-    // allow attributes on module parameters.
-    if (argAttr.first != "firrtl.name")
-      continue;
-
-    return argAttr.second.dyn_cast<StringAttr>();
-  }
-
-  return StringAttr();
+// If the specified module contains the argNames attribute, return it.
+ArrayAttr firrtl::getFIRRTLModuleArgNameAttr(Operation *module) {
+  return module->getAttrOfType<ArrayAttr>("argNames");
```

Comment:
I'd recommend naming the attribute `portNames`.  The RTL dialect splits "arguments" and "results" but in firrtl they are all "ports" which happen to be modeled as arguments.